### ISSUE TITLE
Please evaluate SCL package builds

### ISSUE THREAD

**Issue by Thomas-Gelf** [Icinga/icingaweb2#2847]
I'd like to know:
* whether everything works find on RHEL/CentOS 6/7 using PHP 7.x SCL packages
* what would be required to be adjusted in our packages to build packages depending on SCL
* a short documentation explaining how to enable SCL repos on RHEL and CentOS
* how a related (eventually additional) fpm-only package would look like

Thanks a lot!

**Comment by dgoetz** [comment]
Tested on RHEL 7 (should be the same on the others):
Installation of PHP70 SCL according to https://www.softwarecollections.org/en/scls/rhscl/rh-php70/ and php for testing, should not be required after adjusting the spec.
```
yum install centos-release-scl
yum-config-manager --enable rhel-server-rhscl-7-rpms
yum install rh-php70
yum install rh-php70-php
```

Afterwards you can also install optionally required modules like ldap/mysql/pgsql. This results in the following packages from SCLs (php70 and httpd24) installed:
```
# rpm -qa | grep -e php70 -e httpd24
rh-php70-php-pdo-7.0.10-2.el7.x86_64
rh-php70-php-ldap-7.0.10-2.el7.x86_64
rh-php70-php-json-7.0.10-2.el7.x86_64
rh-php70-php-xml-7.0.10-2.el7.x86_64
httpd24-httpd-tools-2.4.18-11.el7.x86_64
rh-php70-runtime-2.3-1.el7.x86_64
rh-php70-php-zip-7.0.10-2.el7.x86_64
rh-php70-php-cli-7.0.10-2.el7.x86_64
rh-php70-php-process-7.0.10-2.el7.x86_64
rh-php70-2.3-1.el7.x86_64
httpd24-libnghttp2-1.7.1-1.el7.x86_64
httpd24-httpd-2.4.18-11.el7.x86_64
rh-php70-php-gd-7.0.10-2.el7.x86_64
rh-php70-php-intl-7.0.10-2.el7.x86_64
rh-php70-php-mysqlnd-7.0.10-2.el7.x86_64
sclo-php70-php-pecl-imagick-3.4.3-1.el7.x86_64
rh-php70-php-pgsql-7.0.10-2.el7.x86_64
rh-php70-php-common-7.0.10-2.el7.x86_64
rh-php70-php-pear-1.10.1-3.el7.noarch
httpd24-runtime-1.1-14.el7.x86_64
rh-php70-php-7.0.10-2.el7.x86_64
```

Following changes to the SPEC are required:
```
# diff -u ~/rpmbuild/SPECS/icingaweb2.spec.orig ~/rpmbuild/SPECS/icingaweb2.spec
--- /root/rpmbuild/SPECS/icingaweb2.spec.orig	2017-05-22 12:17:50.238638141 +0000
+++ /root/rpmbuild/SPECS/icingaweb2.spec	2017-05-22 11:32:57.936461642 +0000
@@ -15,9 +15,9 @@
 Packager:       Icinga Team <info@icinga.com>
 
 %if 0%{?fedora} || 0%{?rhel} || 0%{?amzn}
-%define php             php
-%define php_cli         php-cli
-%define wwwconfigdir    %{_sysconfdir}/httpd/conf.d
+%define php             rh-php70-php
+%define php_cli         rh-php70-php-cli
+%define wwwconfigdir    /opt/rh/httpd24/root/etc/httpd/conf.d/
 %define wwwuser         apache
 %endif
 
@@ -84,7 +84,7 @@
 Requires:                   %{name}-vendor-zf1 = 1.12.20-1%{?dist}
 %{?amzn:Requires:           %{php}-pecl-imagick}
 %{?fedora:Requires:         php-pecl-imagick}
-%{?rhel:Requires:           php-pecl-imagick}
+%{?rhel:Requires:           sclo-php70-php-pecl-imagick}
 %{?suse_version:Requires:   %{php}-gettext %{php}-json %{php}-openssl %{php}-posix}
 
 %description -n php-Icinga
@@ -100,6 +100,7 @@
 %{?fedora:Requires:         %{php_cli} >= 5.3.0 bash-completion}
 %{?rhel:Requires:           %{php_cli} >= 5.3.0 bash-completion}
 %{?suse_version:Requires:   %{php} >= 5.3.0}
+AutoReq:                    false
 
 %description -n icingacli
 Icinga CLI
```

I think `AutoReq: false` is only required because of hardcoded /usr/bin/php in icingacli, which should be changed to env  because it will not find an interpreter during runtime at this path also after enabling scl with `scl enable rh-php70 bash`. Perhaps we should create a wrapper script so users must not enable scl to use icingacli?

PHP configuration will be in `/etc/opt/rh/rh-php70/php.ini`.

SELinux-Policy is fine for using SCL as it uses local fcontext equivalence.

There is also a package `rh-php70-php-fpm` which is a pretty default fpm, so running Icinga Web 2 with an existing configuration on fpm should only be a matter of adjusting paths.

**Comment by lazyfrosch** [comment]
The problem is AFAIK that the user can not really choose than what PHP to use...

Please don't forget that the `icingaweb2.spec` here will be a reference in the future.

Issues, PR and changes should go to: https://github.com/icinga/icinga-packaging/tree/rpm/snapshot

**Comment by lippserd** [comment]
Will be fixed w/ icinga/icinga-packaging#41.
